### ISSUE TITLE
Winget reports packages in the "**msstore**" with available version 'Unknown'

### ISSUE THREAD

**Issue by littlef0x** [microsoft/winget-cli#1434]
# Brief description of your issue

Some packages in the msstore repo are tagged as version 'Latest', and winget regards them as upgradable all the time. For me, it happens to package Microsoft Whiteboard (Microsoft.Whiteboard) and Python 3.8 (PythonSoftwareFoundation.Python.3.8).
```
> winget upgrade | findstr Latest
Microsoft Whiteboard                         Microsoft.Whiteboard                21.10823.5772.0 Latest         msstore
Python 3.8                                   PythonSoftwareFoundation.Python.3.8 3.8.2800.0      Latest         msstore
```
However, the latest version is actually already installed. So if I try to upgrade it, winget will check for upgrade and then do nothing, although still reports there's an upgrade available after that.
# Steps to reproduce

Take Microsoft Whiteboard as an example. Run the command:
`winget upgrade`
or
`winget upgrade --all`
or
`winget upgrade Microsoft.Whiteboard`
# Expected behavior
Microsoft Whiteboard will not show in the list since I have the latest version already installed.

# Actual behavior
For `winget upgrade`:
```
> winget upgrade
Name                                         Id                                  Version         Available      Source
-----------------------------------------------------------------------------------------------------------------------
Microsoft Whiteboard                         Microsoft.Whiteboard                21.10823.5772.0 Latest         msstore
```
For `winget upgrade -all` or `winget upgrade Microsoft.Whiteboard`:
```
> winget upgrade microsoft.whiteboard
已找到 Microsoft Whiteboard [Microsoft.Whiteboard]
验证/请求程序包获取成功...
验证/请求程序包获取成功
正在启动程序包安装...
找不到适用的更新。
```
Sorry for the inconvenience since my system language is Chinese. The resutls simply means 'I checked the repo successfully and found no upgrade available'.
BTW, these packages were first installed via MS store rather than winget command. But I think there's no difference, isn't it?

# Environment

```
[winget --info]
Windows Package Manager version
Windows: Windows.Desktop v10.0.19043.1165
Package: Microsoft.DesktopAppInstaller v1.12.11692.0
```

**Comment by denelon** [comment]
@littlef0x the store source doesn't currently provide the version for the package. We don't currently have a way to know if the one installed on your system is up to date or not without that information. We are currently working with the store team on a new "source". We will be changing "latest" to "unknown" until such time as the store source can provide the version so the package manager will know if it should update. If you have the Microsoft Store app, it will provide updates as they are released automatically.

**Comment by littlef0x** [comment]
Thanks for your reply. I examed all items listed by `winget list` and indeed, these two packages are the only two tagged as source 'msstore'. It means that most packages installed via Microsoft Store are not identified, although for many of them, Microsoft Store is the only way to install. Furthermore, I found more confusion related to msstore source.

- Some packages in Microsoft Store can be listed in `winget list`, but cannot be found by `winget search`, e.g., [VLC](https://www.microsoft.com/en-us/p/vlc/9nblggh4vvnh).
- Some packages installed via Microsoft Store seems to be upgraded using winget source rather than msstore source, e.g., [Notepads](https://www.microsoft.com/en-us/p/notepads-app/9nhl4nsc67wm). There is a msstore version of the same software with a different ID. I don't know whether they are exactly the same and why they use different IDs (JackieLiu.NotepadsApp vs. 19282JackieLiu.Notepads-Beta), while some other packages use the same ID in different sources, e.g., [Windows Terminal](https://www.microsoft.com/en-us/p/windows-terminal/9n0dx20hk701). This may cause severe confusion.

**Comment by denelon** [comment]
With the latest release, the "**msstore**" source displays "Unknown" for the version. We have also removed the correlation from the store source for list and export due to current matching logic.

I'm modifying the title on this Issue to match the current experience.

**Comment by denelon** [comment]
We've released Windows Package Manager 1.3.

The Microsoft Store still reports "Unknown" in the version. Other Issues are tracking this current state of the system, and work is in progress to begin reporting versions from the Microsoft Store.

As the `winget upgrade --all` experience has been improved to exclude version "Unknown" I'm going to close this bug in favor of others that are tracking progress.

**Comment by bogdan-patraucean** [comment]
@denelon, any update on this?
